### PR TITLE
Dispatch scroll events

### PR DESCRIPTION
**Pull request by DanielApt**
## Feature

Perfect scrollbar will dispatch custom events:
### ps-scroll-y

This event fires when the y-axis is scrolled in either direction.
### ps-scroll-x

This event fires when the x-axis is scrolled in either direction.
### ps-scroll-up

This event fires when scrolling upwards.
### ps-scroll-down

This event fires when scrolling downwards.
### ps-scroll-left

This event fires when scrolling to the left.
### ps-scroll-right

This event fires when scrolling to the right.
## Technical

All modifications of `scrollTop` and `scrollLeft` are done by the new `updateScroll()` function. This is also the function which detects the scroll direction, and dispatches the events. `updateScroll()` can be found in `/src/js/plugin/update-scroll.js`
## Improvements

This addresses issue no. #94

### CODE REVIEW

**Comment by hatashiro** [comment]
Looks superb :thumbsup: Can you add an example too? If you're busy, I can do.

**Comment by hatashiro** [comment]
And I'd like to add you as a contributor of this project, thanks to your wonderful helps. Please feel free to tell me if you like it.

File: README.md [hunk 1]
```diff
@@ -312,6 +312,41 @@ The number of pixels the content height can surpass the container height without
...
+This event fires when scrolling to the left.
+
+### ps-scroll-right
+This event fires when scrolling to the right.
+
+You can listen to these events either with vanilla JavaScript
+```javascript
+document.addEventListener('ps-scroll-x', function () {
+  // ...
+}
```
Comment (hatashiro) [inline README.md]: Maybe

``` javascript
})
```

File: README.md [hunk 2]
```diff
@@ -312,6 +312,41 @@ The number of pixels the content height can surpass the container height without
...
+```
+or with jQuery
+```javascript
+$(document).on('ps-scroll-x', function () {
+  // ...
+}
```
Comment (hatashiro) [inline README.md]:
```
})
```

**Comment by DanielApt** [comment]
Sure thing, I'll make the changes to the documentation, and I'll also add an example.

I'm happy to be added as a contributor, thanks!

**Comment by DanielApt** [comment]
@noraesae Fixed the documentation and added an example.

**Comment by hatashiro** [comment]
Great :smile: I added you as a collaborator too :+1:

**Comment by hatashiro** [comment]
From now on, please feel free to push any commit, resolve any issue and so on. I will also review the work and leave a comment.

Thanks!

**Comment by DanielApt** [comment]
Thanks @noraesae, I actually realised there could also be the need for events when the user hits the top / bottom of the chart. I was initially thinking of calling these `ps-scroll-top` and `ps-scroll-bottom`, but this wouldn't be semantic for events on the x-axis.

Any suggestions for some better naming?

**Comment by hatashiro** [comment]
Til now I've recommended user using the custom callback to handle the event, like [this](https://github.com/noraesae/perfect-scrollbar/blob/master/examples/reach-the-end-callback.html), but I think now we have a custom event interface and there's no reason not to implement that.

I'm also not sure about the event name, but may using `reach` in the names looks okay?

**Comment by zomp** [comment]
You can call it `ps-scroll-start` and `ps-scroll-end` – it is used by CSS' [flexbox](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Flexible_Box_Layout/Using_CSS_flexible_boxes#Flexible_boxes_vocabulary).

**Comment by DanielApt** [comment]
@zomp, good suggestion. I like calling these sides `start` and `end`. Now these events could occur both on y as well as the x axis.

Also `start` or `end` could imply something is starting/ending, but with the usage of the word `reach` this could be solved:

Bringing it all together I think we could use something like:
`ps-x-reach-start`, `ps-x-reach-end` (or `ps-x-reached-start`)?

**Comment by zomp** [comment]
@DanielApt, I like the names. Infinitive `reach` is consistent with `click`, `load`, `focus` and other event names. The order (infinitive + specification) is consistent with `mouseover`, `keydown`...

**Comment by DanielApt** [comment]
Alright, I'll implement the events `ps-x-reach-start`, `ps-x-reach-end`, `ps-y-reach-start` and `ps-y-reach-end`

**Comment by DanielApt** [comment]
Just added the new events as well.

**Comment by hatashiro** [comment]
Great work.

**Pull request merged**